src/main/java/za/ac/sun/cs/search/singleagent/IDAStarAgent.java

```diff
@@ -25,7 +23,7 @@ public IDAStarAgent(short[] configuration) {
      */
     @Override
     public Direction[] solve() {
-        Stack<Direction> path = new Stack<Direction>();
+        Stack<Direction> path = new Stack<>();
```

Comment:
You might want to consider `ArrayDeque`, as stated on the [`Stack` documentation](https://docs.oracle.com/javase/7/docs/api/java/util/Stack.html): 
> A more complete and consistent set of LIFO stack operations is provided by the Deque interface and its implementations, which should be used in preference to this class.

Reply:
I'll look into it. Both provide the functionality needed.